dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/enrollment/EnrollmentsExportController.java

```diff
@@ -120,32 +107,32 @@ ResponseEntity<Page<ObjectNode>> getEnrollments(
       org.hisp.dhis.tracker.Page<Enrollment> page =
           enrollmentsPage.withMappedItems(ENROLLMENT_MAPPER::map);
 
-      return requestHandler.serve(request, ENROLLMENTS, page, requestParams);
+      return new FilteredPage<>(
+          Page.withPager(ENROLLMENTS, page, getRequestURL(request)), requestParams.getFields());
     }
 
     List<Enrollment> enrollments =
         enrollmentService.findEnrollments(operationParams).stream()
             .map(ENROLLMENT_MAPPER::map)
             .toList();
 
-    return requestHandler.serve(ENROLLMENTS, enrollments, requestParams);
+    return new FilteredPage<>(
+        Page.withoutPager(ENROLLMENTS, enrollments), requestParams.getFields());
   }
 
   @OpenApi.Response(OpenApi.EntityType.class)
   @GetMapping(value = "/{uid}")
-  public ResponseEntity<ObjectNode> getEnrollmentByUid(
+  public FilteredEntity<Enrollment> getEnrollmentByUid(
```

Comment:
I like this new way. We will have to patch OpenAPI to understand the `FilteredEntity` properly (in case you have not done it - just doing a quick scroll though).